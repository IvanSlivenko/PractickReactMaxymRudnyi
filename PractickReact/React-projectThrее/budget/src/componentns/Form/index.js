import PropTypes from "prop-types";
import { useState } from "react";
import { Row } from "./styles";
import { Comment } from "./styles";
import { Input } from "./styles";
import { Button } from "./styles";
import { Wrapper } from "./styles";


const Form =(props)=> {

  const [form, setForm]=useState(
    {
      value: '',
      date: new Date().toISOString().substring(0, 10),
      comment:''
    
    }
  )
  

  const onSubmit = (e) => {
    e.preventDefault();

   props.onChange(form);
   
    setForm({
        ...form,
        value: '',
        comment:''
        
      })
  };

  const onChange = (e) => {
    const { value, name } = e.target;
console.log('onchange', e);

    setForm({
      ...form,
      [name]: value
    });
  };

  
    return (
      <Wrapper>
        <form onSubmit={onSubmit}>
          <Row>
            <Input
              name="value"
              type="number"
              placeholder="Сумма"
              value={form.value}
              onChange={onChange}
            />
            <Input
              type="date"
              name="date"
              value={form.date}
              onChange={onChange}
            />
          </Row>

          <Row>
            <Button> Зберегти </Button>
            <Comment
              name="comment"
              value={form.comment}
              onChange={onChange}
            />
          </Row>
        </form>
      </Wrapper>
    );  
}

Form.propTypes = {
  onChange: PropTypes.func,
};
export default Form;
