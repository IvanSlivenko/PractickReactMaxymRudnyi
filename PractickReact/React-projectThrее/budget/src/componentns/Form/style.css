

.styles {
background: #fff;
border: 1px solid #dbdbdb;
border-radius: 3px;
margin-bottom: 10px;

border-radius: 3px;
padding: 0 5px;
min-width: 80px;
margin-bottom: 3px;
margin-right: 10px;
border: 1px solid #dbdbdb;
height: 40px;
outline: none;

display: flex;
align-items: center;


border:  1px solid #dbdbdb;
height: 40px;
border-radius: 3px;
padding: 0 21px;
box-sizing: border-box;
background-color: #e0e0e0;
color:rgba(0, 0, 0, 0.87);
font-size: 18px;
margin-right: 10px;
outline: none;
width: 118px;

/* &:hover{
    cursor: pointer;
} */


min-height: 40px;
line-height: 1.5;
box-sizing: border-box;
padding: 5px;
border-radius: 3px;
width: 138px;
margin: 0px;
height: 40px;
min-width: 140px;
border: 1px solid #dbdbdb;
outline: none;

flex-grow: 1;

display: flex;
align-items: center;
justify-content: space-between;

}