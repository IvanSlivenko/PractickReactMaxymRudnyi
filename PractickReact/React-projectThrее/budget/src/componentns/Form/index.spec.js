import React from "react";
import { shallow } from 'enzyme';
import Form from ".";

describe('Form component', () => {
    let props;
    let sut;

    beforeEach(() => {
        props = {
            onChange: jest.fn()
        }

        sut = shallow(<Form {...props} />);
    })

    describe('when edit form', () => {
        it("should change value input", () => {
       
            const input = sut.find('Input[name="value"]').at(0);
            input.simulate("change", { target: { value: '123' } });
            
            sut.update();

            const { value } = input.props();
            expect(value).toBe('123');
        })          
    })
})