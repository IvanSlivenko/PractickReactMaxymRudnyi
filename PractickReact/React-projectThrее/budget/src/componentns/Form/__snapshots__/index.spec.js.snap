// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fo component should show transaction 1`] = `
<TransactionsWrapper>
  <form
    onSubmit={[Function]}
  >
    <Row>
      <Input
        name="value"
        onChange={[Function]}
        placeholder="Сумма"
        type="number"
        value=""
      />
      <Input
        name="date"
        onChange={[Function]}
        type="date"
        value="2023-09-01"
      />
    </Row>
    <Row>
      <Button>
         Зберегти 
      </Button>
      <Comment
        name="comment"
        onChange={[Function]}
        value=""
      />
    </Row>
  </form>
</TransactionsWrapper>
`;

exports[`Form component should show transaction 1`] = `
<TransactionsWrapper>
  <form
    onSubmit={[Function]}
  >
    <Row>
      <Input
        name="value"
        onChange={[Function]}
        placeholder="Сумма"
        type="number"
        value=""
      />
      <Input
        name="date"
        onChange={[Function]}
        type="date"
        value="2023-09-01"
      />
    </Row>
    <Row>
      <Button>
         Зберегти 
      </Button>
      <Comment
        name="comment"
        onChange={[Function]}
        value=""
      />
    </Row>
  </form>
</TransactionsWrapper>
`;

exports[`Transaction component should show transaction 1`] = `
<TransactionsWrapper>
  <form
    onSubmit={[Function]}
  >
    <Row>
      <Input
        name="value"
        onChange={[Function]}
        placeholder="Сумма"
        type="number"
        value=""
      />
      <Input
        name="date"
        onChange={[Function]}
        type="date"
        value="2023-09-01"
      />
    </Row>
    <Row>
      <Button>
         Зберегти 
      </Button>
      <Comment
        name="comment"
        onChange={[Function]}
        value=""
      />
    </Row>
  </form>
</TransactionsWrapper>
`;
